docs/statistics.md

```diff
@@ -1,269 +1,425 @@
 
-## pktSentTotal
+The total number of data packets detected as lost at the receiver side. Available for receiver.
 
-The total number of sent data packets, including retransmitted packets. Applicable for data sender.
+If a packet is received out of order, the gap (sequence discontinuity) is also taken into account and the number of lost packets is increased by the discontinuity size independently of the reordering tolerance value.
 
-## pktRecvTotal
+Lost packets detection is based on the gaps in sequence numbers of the SRT DATA packets and is triggered by a newly received packet. An offset is calculated between sequence numbers of the newly arrived DATA packet and the previously received DATA packet (the packet with the highest sequence number). Receiving older packets does not affect this value. The packets from that gap are considered lost,
+and that number is added to the `pktRcvLossTotal` measurement. In the case where
+the offset is negative, the packet is considered late, meaning that it was either
+already acknowledged or dropped by TSBPD as too late to be delivered. Such late
+packets are ignored.
 
-The total number of received packets, including received retransmitted packets. Applicable for data receiver.
+TODO: 
 
-## pktSndLossTotal
+- This is wrong -> ticket, correct this and make separate statistic. Decrypted statistic should include wrongly encrypted packets. Here we should save the comment regarding the version 1.4.0-1.4.1 and include the comment that since version 1.5.0 we do not take this into account.
+- Calculate loss for FEC -> ticket (low priority), there is something already
+- We have already pktRcvUndecryptTotal
 
-The total number of data packets considered or reported lost (sender side). Does not correspond to packets detected as lost at receiving side. Applicable for data sender.
+Since SRT v1.4.0, includes packets that failed to be decrypted.
 
-A packet is considered lost in two cases: 
-1. Sender receives a loss report from receiver.
-2. Sender initiates retransmission after not receiving ACK for a certain timeout. Refer to `FASTREXMIT` and `LATEREXMIT` algorithms.
+### pktRetransTotal
 
-## pktRcvLossTotal
+The total number of retransmitted packets. Available for sender.
 
-The total number of data packets detected lost on the receiver's side.
+TODO: 
 
-Includes packets that failed to be decrypted (only as of SRT version 1.4.0).
+- We can calculate retransmitted packets both at the sender side and at the receiver side which is stated in pktSentTotal and pktRecvTotal stats -> we should have this statistics available both for snd and rcv. The same for byteRetransTotal -> rename this to pktSentRetransTotal plus add new statistics pktRecvRetransTotal, be careful this is API change. So, think how we can use two names pktRetransTotal and pktSentRetransTotal, until pktRetransTotal can be deprecated. And here we should put special comment. Create ticket for this.
```

Comment:
Retransmitted packets calculated on the sender count those that departed the sender; those calculated on the receiver count those that arrived at the receiver, which is less by the number of retransmitted packets lost. Sides do not interchange this information anyhow. For example, if you want to know how many retransmitted packets have been "lost again", you need to process stats collected separately on both sides. The difference between `pktRetrans` on sender and `pktRcvRetrans` on receiver is the number of "lost again" packets. Note that `pktRcvRetrans` is a newly introduced stat since the R retransmission flag has been added - without this flag distinction between unordered and retransmitted packets would not be possible.